Source/NSAttributedString.m

```diff
@@ -1095,20 +1096,43 @@ - (NSFileWrapper *) fileWrapperFromRange: (NSRange)range
 - (NSInteger) itemNumberInTextList: (NSTextList *)list
                            atIndex: (NSUInteger)location
 {
-  NSParagraphStyle *style = [self attribute: NSParagraphStyleAttributeName
-                                  atIndex: location
-                                  effectiveRange: NULL];
-  if (style != nil)
+  NSRange listRange = [self rangeOfTextList: list atIndex: location];
+  if (listRange.location == NSNotFound)
     {
-      NSArray *textLists = [style textLists];
+      return 0;
```

Comment:
This matches the return value in Cocoa (I'd have expected `NSNotFound` but it seems to return 0 instead).